pkg/controller/controller_ref_manager.go

```diff
@@ -183,6 +183,10 @@ func (m *PodControllerRefManager) ClaimPods(pods []*v1.Pod, filters ...func(*v1.
 
 	match := func(obj metav1.Object) bool {
 		pod := obj.(*v1.Pod)
+		// Skip pods marked for deletion
```

Comment:
Ah, yes. I've misread how it's working. It looks like I need to add some unit tests here. I will move it outside of match function.

Please note that you are filtering inactive pods. I'm filtering pods marked for deletion which in some cases may still be active. I think it's different use case.